clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -694,7 +694,12 @@ void CodeGenFunction::EmitOpenCLKernelMetadata(const FunctionDecl *FD,
 
   if (const SYCLIntelNoGlobalWorkOffsetAttr *A =
           FD->getAttr<SYCLIntelNoGlobalWorkOffsetAttr>()) {
-    if (A->getEnabled())
+    const Expr *Arg = A->getValue();
+    assert(Arg && "Got an unexpected null argument");
+    Optional<llvm::APSInt> ArgVal =
+        Arg->getIntegerConstantExpr(FD->getASTContext());
+    assert(ArgVal.hasValue() && "Not an integer constant expression");
+    if (ArgVal)
```

Comment:
This is testing whether the `Optional<>` has a value or not (which we assert to be true just one line above). This should be `if (ArgVal->getBoolValue())` to look through the `Optional<>` and query the value of the underlying `APSInt`.

Reply:
Done.